### PR TITLE
Rename Starlite

### PR DESCRIPTION
**Pull request by Goldziher**
This PR renames:

 `starlite` -> `litestar`, `Starlite` -> `Litestar`, `STARLITE` -> `LITESTAR`

It does not affect changes to:

1. urls
2. github workflows and some other .github components
3. README.md
4. docs/release-notes.rst

Additionally, it temporarily turns off the deployment of docs.

Finally, it fixes the validation of Brotli imports, since it appeared as a sonar issue (we should start looking at these in PRs more carefully)

### CODE REVIEW

File: docs/release-notes/changelog.rst [hunk 1]
```diff
@@ -58,7 +58,7 @@
         :pr: 1277
 
         Add a ``__version__`` constant to the ``starlite`` namespace, containing a
-        :class:`NamedTuple <typing.NamedTuple>`, holding information about the currently
+        ``starlitetyping.NamedTuple``, holding information about the currently
```
Comment (provinzkraut) [inline docs/release-notes/changelog.rst:61]: What happened here? Regex gone wild? :wink: 
```suggestion
        :class:`NamedTuple <typing.NamedTuple>`, holding information about the currently
```

File: docs/release-notes/changelog.rst [hunk 2]
```diff
@@ -69,30 +69,27 @@
...
 
         Update the CLI :ref:`usage/cli:autodiscovery` to only consider canonical modules app and application, but every
-        :class:`Starlite <.app.Starlite>` instance or application factory able to return a ``Starlite`` instance within
+        ``starlite.app.Starlite`` instance or application factory able to return a ``Starlite`` instance within
         those or one of their submodules, giving priority to the canonical names app and application for application
         objects and submodules containing them.
 
         .. seealso::
-            :ref:`CLI autodiscovery <usage/cli:autodiscovery>`
+            :ref``starliteusage/cli:autodiscovery``
```
Comment (provinzkraut) [inline docs/release-notes/changelog.rst:84]:
```suggestion
            :ref:`CLI autodiscovery <usage/cli:autodiscovery>`
```

File: docs/release-notes/changelog.rst [hunk 3]
```diff
@@ -417,10 +414,10 @@
 
         :ref:`usage/responses:Response Headers` and :ref:`usage/responses:Response Cookies` now have the same
         interface, along with the ``headers`` and ``cookies`` keyword arguments to
-        :class:`Response <starlite.response.Response>`. They each allow to pass either a
-        :class:`Mapping[str, str] <typing.Mapping>`, e.g. a dictionary, or a :class:`Sequence <typing.Sequence>` of
-        :class:`ResponseHeaders <starlite.datastructures.response_header.ResponseHeader>` or
-        :class:`Cookies <starlite.datastructures.cookie.Cookie>` respectively.
+        ``starlite.response.Response``. They each allow to pass either a
+        ``Mapping[st``starlitetyping.Mapping``, e.g. a dictionary, or a ``starlitetyping.Sequence`` of
```
Comment (provinzkraut) [inline docs/release-notes/changelog.rst:418]:
```suggestion
        ``starlite.response.Response``. They each allow to pass either a
        `:class:`Mapping[str, str] <typing.Mapping>`, e.g. a dictionary, or a :class:`Sequence <typing.Sequence>` of
```

File: docs/release-notes/changelog.rst [hunk 4]
```diff
@@ -479,14 +476,12 @@
...
         the time it takes to ``import starlite``.
-        An overview of the changed import paths can be found in the
-        :ref:`migration guide <release-notes/migration_guide_2:Changed module paths>`
 
     .. change:: Fix resolving of relative paths in ``StaticFilesConfig``
         :type: bugfix
         :pr: 1256
 
-        Using a relative :class:`pathlib.Path` did not resolve correctly and result in a ``NotFoundException``
+        Using a relative ``pathlib.Path` did not resolve correctly and result in a ``NotFoundException``
```
Comment (provinzkraut) [inline docs/release-notes/changelog.rst:484]:
```suggestion
        Using a relative :class:`pathlib.Path` did not resolve correctly and result in a ``NotFoundException``
```

File: docs/release-notes/changelog.rst [hunk 5]
```diff
@@ -205,7 +202,7 @@
     .. change:: Remove ``picologging`` dependency from the ``standard`` package extra
         :type: misc
         :breaking:
-        :pr: 1313
+        :pr: 131
```
Comment (provinzkraut) [inline docs/release-notes/changelog.rst:205]:
```suggestion
        :pr: 1313
```

**Review by provinzkraut — COMMENTED** [review]
Found a few more issues in the changelog

**Review by peterschutt — APPROVED** [review]

**Pull request merged**